packages/yoroi-extension/app/UI/features/staking/useCases/RewardsSummary/RewardHistoryGraph.tsx

```diff
@@ -0,0 +1,205 @@
+  objectFit: 'scale-down',
+});
+
+const AvatarWrapper = styled(Box)({
+  width: '24px',
+  height: '24px',
+  minWidth: '24px',
+  marginRight: '12px',
+  borderRadius: '20px',
+  overflow: 'hidden',
+});
+
+const ExpandMoreIcon: React.FC = () => (
+  <svg width="14" height="8" viewBox="0 0 14 8" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <path
+      fillRule="evenodd"
+      clipRule="evenodd"
+      d="M0.292893 0.292893C0.683417 -0.0976311 1.31658 -0.0976311 1.70711 0.292893L7 5.58579L12.2929 0.292893C12.6834 -0.0976311 13.3166 -0.0976311 13.7071 0.292893C14.0976 0.683417 14.0976 1.31658 13.7071 1.70711L7.70711 7.70711C7.31658 8.09763 6.68342 8.09763 6.29289 7.70711L0.292893 1.70711C-0.0976311 1.31658 -0.0976311 0.683417 0.292893 0.292893Z"
+      fill="#6B7384"
+    />
+  </svg>
+);
+
+const AccordionSummary = styled((props: MuiAccordionSummaryProps) => (
+  <MuiAccordionSummary expandIcon={<ExpandMoreIcon />} {...props} />
+))(() => ({
+  padding: 0,
+  '.MuiAccordionSummary-content': {
+    margin: 0,
+  },
+  '& .MuiAccordionSummary-expandIconWrapper.Mui-expanded': {
+    transform: 'rotate(180deg)',
+  },
+  '& .MuiCollapse-vertical': {
+    transitionDuration: '0.5s',
+  },
+}));
+
+const RewardHistoryGraph: React.FC<RewardHistoryGraphProps> = ({ graphData, onOpenRewardList }) => {
+  const strings = useStrings();
+  const { rewardsGraphData } = graphData;
+  const rewardList = rewardsGraphData.items?.perEpochRewards;
+  const title = strings.rewardHistoryLabel;
+
+  return (
+    <Box
+      p="24px"
+      sx={{
+        display: 'flex',
+        flexFlow: 'column',
+        justifyContent: 'space-between',
+      }}
+    >
+      <Box
+        sx={{
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'space-between',
+          width: '100%',
+        }}
+      >
+        <Typography component="div" variant="body1" fontWeight={500} color="ds.text_gray_medium">
+          {title}
+        </Typography>
+        <Button
+          // @ts-ignore
+          variant="primary"
+          size="medium"
+          onClick={onOpenRewardList}
+          sx={{ lineHeight: '21px' }}
+        >
+          {title}
+        </Button>
+      </Box>
+
+      {rewardsGraphData.error && !rewardsGraphData.items && (
+        <div>
+          <Typography variant="body2" color="ds.text_error">
+            {strings.errorLabel}
+          </Typography>
+        </div>
+      )}
+
+      {!Array.isArray(rewardList) ? (
+        <CircularProgress />
+      ) : (
+        <Box ml="-50px">
+          <RewardGraphClean
+            epochTitle={strings.epochLabel}
+            stakepoolNameTitle={strings.stakepoolNameLabel}
+            xAxisLabel={strings.epochLabel}
+            yAxisLabel={strings.rewardValue}
+            primaryBarLabel={strings.rewardsLabel}
+            data={rewardList}
+            hideYAxis={rewardsGraphData.hideYAxis}
+          />
+        </Box>
+      )}
+    </Box>
+  );
+};
+
```

Comment:
maybe better to import these from another local `components` file, just to clean up a bit and make component smaller, nothing major